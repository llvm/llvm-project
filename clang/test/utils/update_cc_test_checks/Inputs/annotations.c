// RUN: %clang_cc1 -triple x86_64-unknown-unknown -fblocks -ftrivial-auto-var-init=zero %s -emit-llvm -o - | FileCheck %s

int foo() {
    int x = x + 1;
    return x;
}
