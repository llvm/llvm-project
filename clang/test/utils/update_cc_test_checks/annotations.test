## Test that !annotation metadata is matched correctly

# RUN: cp %S/Inputs/annotations.c %t.c && %update_cc_test_checks --version 4 %t.c
# RUN: diff -u %S/Inputs/annotations.c.expected %t.c
