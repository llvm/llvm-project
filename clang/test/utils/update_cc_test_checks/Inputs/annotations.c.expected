// NOTE: Assertions have been autogenerated by utils/update_cc_test_checks.py UTC_ARGS: --version 4
// RUN: %clang_cc1 -triple x86_64-unknown-unknown -fblocks -ftrivial-auto-var-init=zero %s -emit-llvm -o - | FileCheck %s

// CHECK-LABEL: define dso_local i32 @foo(
// CHECK-SAME: ) #[[ATTR0:[0-9]+]] {
// CHECK-NEXT:  entry:
// CHECK-NEXT:    [[X:%.*]] = alloca i32, align 4
// CHECK-NEXT:    store i32 0, ptr [[X]], align 4, !annotation [[META2:![0-9]+]]
// CHECK-NEXT:    [[TMP0:%.*]] = load i32, ptr [[X]], align 4
// CHECK-NEXT:    [[ADD:%.*]] = add nsw i32 [[TMP0]], 1
// CHECK-NEXT:    store i32 [[ADD]], ptr [[X]], align 4
// CHECK-NEXT:    [[TMP1:%.*]] = load i32, ptr [[X]], align 4
// CHECK-NEXT:    ret i32 [[TMP1]]
//
int foo() {
    int x = x + 1;
    return x;
}
//.
// CHECK: [[META2]] = !{!"auto-init"}
//.
